### PR TITLE
Add preview endpoints command

### LINKED ISSUE #1665: Support for "okteto preview endpoints"
**Is your feature request related to a problem? Please describe.**
I want to see the endpoints of my preview environment.

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Fixes #1665

## Proposed changes

- Adds okteto preview endpoints command
-

### CODE REVIEW

**Review by ifbyol — APPROVED** [review]

**Review by rberrelleza — COMMENTED** [review]
one common use case I can see for this command is to use it as part of an automated script (e.g. get the endpoint and then run tests, or send it to slack). have we considered an option for having a different format? 

Also, could we integrate this with our current github action for preview envs? I think it would be better if the action used this instead of the custom go code (from a maintainability perspective)

**Review by pchico83 — APPROVED** [review]

**Comment by jLopezbarb** [comment]
using this code instead of the one we have in the action will result in not being able to create a hyperlink to each of them since you need to use markdown syntax

**Comment by rberrelleza** [comment]
Then what is the use case for this command? Is this for people to run it locally on their own machines?

**Comment by jLopezbarb** [comment]
I think it can be useful for Gitlab actions where we don't write any comment on the PR

File: cmd/preview/preview.go [hunk 1]
```diff
@@ -34,8 +34,12 @@ func Preview(ctx context.Context) *cobra.Command {
 	listCommand := List(ctx)
 	listCommand.Hidden = true
 
+	endpointsCommand := Endpoints(ctx)
+	endpointsCommand.Hidden = true
```
Comment (pchico83) [inline cmd/preview/preview.go:38]: Let's show all info about `previews`

File: pkg/okteto/preview.go [hunk 2]
```diff
@@ -136,6 +143,42 @@ func ListPreviews(ctx context.Context) ([]PreviewEnv, error) {
 	return body.Previews, nil
 }
 
+func ListPreviewsEndpoints(ctx context.Context, previewName string) ([]string, error) {
+	endpoints := make([]string, 0)
+
+	q := fmt.Sprintf(`query{
+		preview(id: "%s"){
+			deployments{
```
Comment (pchico83) [inline pkg/okteto/preview.go:151]: You need to query `gitDeploys`, `stacks` and `apps` too, and for each of them, `deployments` and `statefulsets` endpoints:
https://github.com/okteto/app/blob/master/backend/pkg/graphql/types.go#L277

File: pkg/okteto/preview.go [hunk 3]
```diff
@@ -136,6 +143,42 @@ func ListPreviews(ctx context.Context) ([]PreviewEnv, error) {
...
+		}
+	}`, previewName)
+	var body PreviewBody
+	if err := query(ctx, q, &body); err != nil {
+		return nil, err
+	}
+
+	for _, d := range body.Preview.Deployments {
+		for _, endpoint := range d.Endpoints {
+			endpoints = append(endpoints, endpoint.URL)
```
Comment (pchico83) [inline pkg/okteto/preview.go:170]: @maroshii how is the Okteto UI sorting the endpoints?
We should follow the same logic here

**Review by pchico83 — COMMENTED** [review]
@jLopezbarb Let's have a `-o=json` flag to show the endpoints. This way we can use that option to parse the output in the `deploy-preview` action. This is something that people might need to parse the output in other CI tools

File: pkg/okteto/preview.go (hunk 3 above)
Comment (ifbyol) [reply pkg/okteto/preview.go:170]: @pchico83  In the UI endpoints endpoints are sorted by length. Shorter endpoints first. I implemented that logic on the deploy preview github action: https://github.com/okteto/deploy-preview/blob/master/message.go#L174

Comment (pchico83) [reply pkg/okteto/preview.go:170]: @jLopezbarb let's add that logic here too

**Pull request merged**